saskatoon/member/utils.py

```diff
@@ -14,3 +23,72 @@ def reset_password(user: AuthUser) -> str:
     user.has_temporary_password = True
     user.save()
     return password
+
+
+@typechecked
+def _valid_date_contract(_) -> bool:
+    """ If any of these result in an OverflowError then the
+          the condition is broken. We catch the exception so
+          that Deal can report the contract violation correctly """
+    try:
+        _.start - _.buffer
+        _.end + _.buffer
+
+        return True
+    except Exception:
+        return False
+
+
+@deal.chain(
+    deal.pre(lambda _: _.start < _.end,
+             message='end must be later then start'),
+    deal.pre(_valid_date_contract,
+             message='Substracting the buffer from the start date and '
+             'adding the buffer to the end date must result in valid dates'),
+    deal.pre(lambda _: _.start.tzinfo is not None and _.end.tzinfo is not None,
+             message='start and end must be offset aware'))
+@typechecked
+def available_equipment_points(
+    start: datetime,
+    end: datetime,
+    harvest: Optional[Harvest],
+    buffer: timedelta = timedelta(hours=1)
+) -> QuerySet[Organization]:
+    """List all available equipment points for a given datetime range"""
+    try:
+        """ A buffer gives pick leaders a bit of leeway in picking up and returning
+             the equipment, since some harvest sites can be further away """
```

Comment:
This syntax is reserved to docstrings. See [Block Comments](https://peps.python.org/pep-0008/#block-comments)
```suggestion
        # A buffer gives pick leaders a bit of leeway in picking up and returning
        # the equipment, since some harvest sites can be further away
```
Please fix throughout this PR